rust/worker/src/execution/worker_thread.rs

```diff
@@ -50,7 +51,11 @@ impl Component for WorkerThread {
 #[async_trait]
 impl Handler<TaskMessage> for WorkerThread {
     async fn handle(&mut self, task: TaskMessage, ctx: &ComponentContext<WorkerThread>) {
-        task.run().await;
+        // Execute the task with the caller span
+        let parent_id = (*task).getTracingContext();
```

Comment:
If we move this to the system, could we make the creation of a context automatically part of the ComponentExecutors call to handle? That way child spans in actors get their parents "for free"?